src/EditorFeatures/Core/Shared/Tagging/EventSources/AbstractTaggerEventSource.cs

```diff
@@ -3,12 +3,15 @@
 // See the LICENSE file in the project root for more information.
 
 using System;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
 using Microsoft.CodeAnalysis.Editor.Tagging;
 
 namespace Microsoft.CodeAnalysis.Editor.Shared.Tagging
 {
     internal abstract class AbstractTaggerEventSource : ITaggerEventSource
     {
+        private bool _paused;
```

Comment:
We didn't want to disconnect entirely? On one hand I don't know if it really hurts much to stay connected, but given we already had a Connect/Disconnect pair, do we need a second pair?

Reply:
So the contract of Disconnect is that it's somewhat like Dispose.  It's can fundamentally put the event source into an unusable state (and is doc'ed as such).  I suppose we could actually try to make it so that you could repeatedly call Connect/Disconnect... but i'm warier of that.